src/main/java/org/dbos/apiary/mysql/MysqlContext.java

```diff
@@ -45,19 +50,28 @@ private void prepareStatement(PreparedStatement ps, Object[] input) throws SQLEx
                 ps.setString(i + 1, (String) o);
             } else if (o instanceof Long)  {
                 ps.setLong(i + 1, (Long) o);
+            } else if (o instanceof Float)  {
+                ps.setFloat(i + 1, (Float) o);
+            } else if (o instanceof Double) {
+                ps.setDouble(i + 1, (Double) o);
+            } else if (o instanceof Timestamp) {
+                ps.setTimestamp(i + 1, (Timestamp) o);
             } else {
+                logger.info("type {} for input {} not recogized ", o.toString(), i);
                 assert (false); // TODO: More types.
             }
         }
     }
 
     public void executeUpsert(String tableName, String id, Object... input) throws Exception {
+        long t0 = System.nanoTime();
```

Comment:
Let's discuss -- this code does not execute the query and doesn't allow read your own writes.